### PR TITLE
docs: Ensure the CI passes for docs

### PR DESCRIPTION
**Pull request by lexnv**
This PR ensures that the CI step for docs will pass properly.
- use absolute crate paths for some  links
- downgrade links of `pub(super)` to simple mentions

Closes https://github.com/paritytech/cumulus/issues/2665

// @paritytech/subxt-team

### CODE REVIEW

File: client/relay-chain-rpc-interface/src/reconnecting_ws_client.rs [hunk 1]
```diff
@@ -161,7 +161,7 @@ impl ReconnectingWsClient {
 	}
 }
 
-/// Worker that should be used in combination with [`RelayChainRpcClient`]. Must be polled to distribute header notifications to listeners.
+/// Worker that should be used in combination with [`crate::RelayChainRpcClient`]. Must be polled to distribute header notifications to listeners.
```
Comment (davxy) [inline client/relay-chain-rpc-interface/src/reconnecting_ws_client.rs:164]:
```suggestion
/// Worker that should be used in combination with [`crate::RelayChainRpcClient`].
///
/// Must be polled to distribute header notifications to listeners.
```

**Review by niklasad1 — APPROVED** [review]

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable
**Logs**: https://gitlab.parity.io/parity/mirrors/cumulus/-/jobs/2930399

**Comment by niklasad1** [comment]
bot rebase

**Comment by lexnv** [comment]
The failure is due to https://github.com/paritytech/cumulus/pull/2660, I'll update polkadot dep and see if build passes

**Comment by niklasad1** [comment]
I merged that PR, let's see if rebasing again works :pray:

**Comment by niklasad1** [comment]
bot merge

**Pull request merged**